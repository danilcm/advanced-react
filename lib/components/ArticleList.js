import React from 'react';

import ArticleItem from './ArticleItem';

const ArticleList = (props) => {
  const { articles } = props;



  return (
    <div>
      {Object.values(articles).map(article => <ArticleItem 
        key={article.id}
        article={article}
      />)}
    </div>
  );
};

export default ArticleList;

