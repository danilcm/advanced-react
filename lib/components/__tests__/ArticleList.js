import React from 'react';

import ArticleList from '../ArticleList';

import Enzyme, { shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

import enzymeToJson from 'enzyme-to-json';

// Setup enzyme's react adapter
Enzyme.configure({ adapter: new EnzymeAdapter() });


describe('ArticleList', () => {

  const testProps = {
    articles: {
      a: {
        id: 'a',
        body: 'body',
        title: 'title-a',
        date: new Date(100).toISOString()
      },
      b: {
        id: 'b',
        body: 'body',
        title: 'title-b',
        date: new Date(100).toISOString()
      }
    },
    store: {
      getAuthor: jest.fn(() => ({}))
    }
  };

  let tree;


  beforeEach(() => {
    tree = shallow(<ArticleList {...testProps} />);
  });

  it('should match snapshot', () => {
    expect(enzymeToJson(tree)).toMatchSnapshot();    
  });

  it('passes props to ArticleItem correctly', () => {
    const items = tree.getElement().props.children;

    const articlesAsArray = Object.values(testProps.articles);

    items.forEach((node, index) => {
      expect(node.props.article).toEqual(articlesAsArray[index]);
    });
  });
  
  
});
