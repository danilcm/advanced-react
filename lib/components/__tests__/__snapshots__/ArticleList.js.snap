// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList should match snapshot 1`] = `
<div>
  <ArticleWithStore
    article={
      Object {
        "body": "body",
        "date": "1970-01-01T00:00:00.100Z",
        "id": "a",
        "title": "title-a",
      }
    }
    key="a"
  />
  <ArticleWithStore
    article={
      Object {
        "body": "body",
        "date": "1970-01-01T00:00:00.100Z",
        "id": "b",
        "title": "title-b",
      }
    }
    key="b"
  />
</div>
`;
